client/rest-high-level/src/main/java/org/elasticsearch/client/license/StartTrialResponse.java

```diff
@@ -0,0 +1,189 @@
+ * under the License.
+ */
+
+package org.elasticsearch.client.license;
+
+import org.elasticsearch.common.ParseField;
+import org.elasticsearch.common.collect.Tuple;
+import org.elasticsearch.common.xcontent.ConstructingObjectParser;
+import org.elasticsearch.common.xcontent.XContentParseException;
+import org.elasticsearch.common.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.elasticsearch.common.xcontent.ConstructingObjectParser.constructorArg;
+import static org.elasticsearch.common.xcontent.ConstructingObjectParser.optionalConstructorArg;
+
+public class StartTrialResponse {
+
+    private static final ParseField ACKNOWLEDGED_FIELD = new ParseField("acknowledged");
+    private static final ParseField TRIAL_WAS_STARTED_FIELD = new ParseField("trial_was_started");
+    private static final ParseField LICENSE_TYPE_FIELD = new ParseField("type");
+    private static final ParseField ERROR_MESSAGE_FIELD = new ParseField("error_message");
+    private static final ParseField ACKNOWLEDGE_DETAILS_FIELD = new ParseField("acknowledge");
+    private static final ParseField ACKNOWLEDGE_HEADER_FIELD = new ParseField("message");
+
+    private static final ConstructingObjectParser<StartTrialResponse, Void> PARSER = new ConstructingObjectParser<>(
+        "start_trial_response",
+        true,
+        (Object[] arguments, Void aVoid) -> {
+            final boolean acknowledged = (boolean) arguments[0];
+            final boolean trialWasStarted = (boolean) arguments[1];
+            final String licenseType = (String) arguments[2];
+            final String errorMessage = (String) arguments[3];
+
+            @SuppressWarnings("unchecked")
+            final Tuple<String, Map<String, String[]>> acknowledgeDetails = (Tuple<String, Map<String, String[]>>) arguments[4];
+            final String acknowledgeHeader;
+            final Map<String, String[]> acknowledgeMessages;
+
+            if (acknowledgeDetails != null) {
+                acknowledgeHeader = acknowledgeDetails.v1();
+                acknowledgeMessages = acknowledgeDetails.v2();
+            } else {
+                acknowledgeHeader = null;
+                acknowledgeMessages = null;
+            }
+
+            return new StartTrialResponse(acknowledged, trialWasStarted, licenseType, errorMessage, acknowledgeHeader,
+                acknowledgeMessages);
+        }
+    );
+
+    static {
+        PARSER.declareBoolean(constructorArg(), ACKNOWLEDGED_FIELD);
+        PARSER.declareBoolean(constructorArg(), TRIAL_WAS_STARTED_FIELD);
+        PARSER.declareString(optionalConstructorArg(), LICENSE_TYPE_FIELD);
+        PARSER.declareString(optionalConstructorArg(), ERROR_MESSAGE_FIELD);
+        // todo consolidate this parsing with the parsing in PutLicenseResponse
+        PARSER.declareObject(optionalConstructorArg(), (parser, aVoid) -> {
```

Comment:
I dont know a ton about this data structure here, but is it possible to use the parser's built in `.map` functions to return a `Map<String, Object>` ?

Reply:
I think so, I'm not super knowledgeable about the parser facility. I'll give it a shot

Reply:
It looks like `XContentFactory#map` works - somehow that version seems even uglier than the old school parsing code though, let me know if that's not what you meant.

It's a little tricky because the object looks like this, where the header ("message") is parsed as part of the map but is its own particular field that we pull out

```json
{
  "acknowledge": {
    "message": "you need to acknowledge blah blah",
    "security": [ "security won't be enabled blah blah" ],
    "watcher": [ "another acknowledge message "],
  }
}
```

Reply:
HAH, ill have a look.

Reply:
I think that its a bit nicer, given that you cant actually parse something incorrectly. Sure its manipulating a bunch of k/v stuff in that map, but the actual parsing logic is way trappier and more subject to fail if slightly messed up. mucking with a map of list of lists is pretty low from a trappy perspective.